### PR TITLE
Allow spaces in option specification.

### LINKED ISSUE #57: Adding options with space causes program crash
Hello, I am about to ask a question, by raising this issue.

## Summary
- add_options() crashes when cla option definition string has a space in it, and want to know if this is intended.

## Environment
- VS2015 on Windows 7 x64
- Debug(x86) is current project configuration

## How to Reproduce This
```C++
cxxopts::Options options(argv[0], "Program for cxxopts");

options.add_options()
    ("d, debug", "Enables debug")
; // Note the space after "d,"
```

This causes crash at `cxxopts.hpp:1018`: `cxxopts::invalid_option_format_error`
```C++
const auto& short_match = result[2];
```

## Why Do I Raise This Issue
- I am new to C++ CLA library, and not sure if no whitespace between short/long option is allowed.
   - Is this a known convention when writing a CLA option?
- Since I am also new to C++ regex handling, thus not sure if I can handle this issue such fast.

### PR DESCRIPTION
**Pull request by shivakar**
Fixes #57.

---
This change is [Reviewable](https://reviewable.io/reviews/jarro2783/cxxopts/58)

### CODE REVIEW

File: include/cxxopts.hpp [hunk 1]
```diff
@@ -884,7 +884,7 @@ namespace cxxopts
       ("--([[:alnum:]][-_[:alnum:]]+)(=(.*))?|-([[:alnum:]]+)");
 
     std::basic_regex<char> option_specifier
-      ("(([[:alnum:]]),)?([[:alnum:]][-_[:alnum:]]*)?");
+      ("(([[:alnum:]]),)?[ ]*([[:alnum:]][-_[:alnum:]]*)?");
```
Comment (BrendanLeber) [inline include/cxxopts.hpp]: Why not use "\s" for matching whitespace?

Comment (shivakar) [inline include/cxxopts.hpp]: "\s" will match any whitespace character including "\t" and "\n".
I just opted for the tightest implementation for the requirement of allowing spaces.
I don't mind sending another pull request with \s.

Comment (BrendanLeber) [inline include/cxxopts.hpp]: True but I feel it is more clear with what is intended.  Looking at "[ ]" I'm not sure if there is something missing or not.  Would it break anything if someone used tabs or newlines?  Their code would look strange but that's probably the least of their problems.

Comment (shivakar) [inline include/cxxopts.hpp]: Reasoning sounds good to me. Will update the pull request shortly.

**Comment by jarro2783** [comment]
I actually would have preferred just a literal space. I can't see someone writing "f,\nfile" for an option name. You're also not going to write
```
R"*(f
file)*"
```

**Comment by shivakar** [comment]
Do you mean:
1) "(([[:alnum:]]),)?[ ]*([[:alnum:]][-_[:alnum:]]*)?" or
2) "(([[:alnum:]]),)? *([[:alnum:]][-_[:alnum:]]*)?"

I think 1 is more clear.
Do you want an update to this PR or a new PR?

**Comment by jarro2783** [comment]
I think it's clearer with `[ ]`, a lone space is a bit weird. Same PR is fine.

**Comment by shivakar** [comment]
Would it be possible for you to make a new release as well when you merge this PR? I am using CMake to pull a git tag version. Thanks.

**Comment by jarro2783** [comment]
:lgtm:

---

Reviewed 2 of 3 files at r1, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[test/options.cpp, line 67 at r3](https://reviewable.io:443/reviews/jarro2783/cxxopts/58#-Kp8mQ6Z8yPcNNmcuW8S:-Kp8mQ6Z8yPcNNmcuW8T:boddnjr) ([raw file](https://github.com/jarro2783/cxxopts/blob/89a9f215533045fbd745271bba7c650e43fca646/test/options.cpp#L67)):*
> ```cpp
>     "b",
>     "-6",
>     "-p"
> ```

I'll add a test that the "space" option works too, but I'll do that now and merge it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/jarro2783/cxxopts/58#-:-Kp8maLpzff9JrECrW_h:bnfp4nl)*

**Comment by jarro2783** [comment]
I made a tag v1.3.0 for the updated version.

**Pull request closed without merging**